### PR TITLE
fix(deposit): payment method duration badge

### PR DESCRIPTION
**Pull request by wachunei**
## **Description**

This pull request enhances the `BuildQuote` component by improving how payment method durations are displayed and tested. The updates ensure that the correct duration is shown for the selected payment method, and that no duration is shown when no payment method is selected. Additionally, new tests have been added to verify this behavior.

**Improvements to payment method duration display:**

* The `BuildQuote` component now conditionally displays the payment duration tag only if a payment method is selected, and dynamically shows the correct duration based on the selected method.

**Testing enhancements:**

* Added a mock payment method (`MOCK_SEPA_BANK_TRANSFER_PAYMENT_METHOD`) to the test utilities for more comprehensive testing.
* Added tests to verify that the correct duration is shown for the selected payment method and that no duration is shown when no payment method is selected.

## **Changelog**

CHANGELOG entry: Fixed a bug displaying wrong duration for Deposit payment methods

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/TRAM-2738

## **Manual testing steps**

```gherkin
Feature: Deposit

  Scenario: user selects a bank transfer payment method
    Given use visits the Deposit flow

    When user selects a Bank transfer payment method
    Then the right duration value shows in the UI
```

### **Before**

pm_badge_before

### **After**

pm_badge_after

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> Make the payment method duration badge show the selected method’s duration and hide it when no method is selected, with tests and snapshots updated.
> 
> - **BuildQuote UI (`app/components/UI/Ramp/Deposit/Views/BuildQuote/BuildQuote.tsx`)**:
>   - **Payment Duration Badge**: Render duration tag only when `selectedPaymentMethod` exists; label now sourced from `deposit.payment_duration.${selectedPaymentMethod.duration}` instead of hardcoded `instant`.
> - **Tests (`BuildQuote.test.tsx`)**:
>   - Add `MOCK_SEPA_BANK_TRANSFER_PAYMENT_METHOD` for duration scenarios.
>   - Add tests to verify correct duration display and no badge when `selectedPaymentMethod` is `null`.
>   - Update snapshots to reflect new conditional rendering and dynamic labels.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a0a069a7d6e1933206e4d1b6cc8f1f9aff698449. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by wachunei** [comment]
bugbot run

File: app/components/UI/Ramp/Deposit/Views/BuildQuote/BuildQuote.test.tsx [hunk 1]
```diff
@@ -264,6 +265,28 @@ describe('BuildQuote Component', () => {
...
+          selectedPaymentMethod: {
+            ...MOCK_SEPA_BANK_TRANSFER_PAYMENT_METHOD,
+          },
+        }),
+      );
+      render(BuildQuote);
+      expect(screen.toJSON()).toMatchSnapshot();
+    });
+
+    it.only('does not show the duration when selected payment method is null', () => {
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/Ramp/Deposit/Views/BuildQuote/BuildQuote.test.tsx]: aaah yeah, sorry

[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/Ramp/Deposit/Views/BuildQuote/BuildQuote.test.tsx]: bugbot run again

**Review by amitabh94 — APPROVED** [review]
LGTM!

File: app/components/UI/Ramp/Deposit/Views/BuildQuote/BuildQuote.tsx [hunk 2]
```diff
@@ -604,12 +604,16 @@ const BuildQuote = () => {
...
+                {selectedPaymentMethod ? (
+                  <TagBase
+                    includesBorder
+                    textProps={{ variant: TextVariant.BodySM }}
+                  >
+                    {strings(
+                      `deposit.payment_duration.${selectedPaymentMethod.duration}`,
+                    )}
+                  </TagBase>
+                ) : null}
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/Ramp/Deposit/Views/BuildQuote/BuildQuote.tsx:616]: This is a great catch, although it is expected behavior at the moment.
